client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
> I'll also dispatch an action to remove the image from the state whenever the date has passed (from inside a check in `<Gravatar />`).

`<Gravatar />` is currently a really simple component. Is it worth adding some dependency on the data store for this, or is there someplace else this "cache flush" can happen?